src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -43,51 +46,68 @@ public interface IConnectionManager : IDisposable
 
         string GetStats();
 
+        /// <summary>
+        /// Initializes peer connection by creating the peer connectors
+        /// as well as the peer discovery loop.
+        /// </summary>
+        void Initialize();
+
+        /// <summary>Logger factory to create loggers.</summary>
+        ILoggerFactory LoggerFactory { get; }
```

Comment:
This is a problematic approach, now the IConnectionManager has a dependency on ILoggerFactory, IAsyncLoopFactory,INodeLifetime  etc...

They should be hidden away from the interface.

Reply:
I have fixed these dependencies by DI. See the new commit 👍